stingray/largememory.py

```diff
@@ -0,0 +1,558 @@
+
+            if segment_size < ev1.time.size / 4096:
+                warnings.warn(
+                    f"It is advisable to have the segment size greater than or equal to {ev1.time.size / 4096}. Very small segment sizes may greatly increase computation times."
+                )
+
+            avg_cspec = stingray.AveragedCrossspectrum(data1=ev1, data2=ev2, segment_size=ev1.time.size / segment_size, norm=norm, gti=gti, power_type=power_type, silent=silent, dt=dt1, large_data=False)
+
+            fin_spec = _addSpectra(fin_spec, avg_cspec, flag)
+
+        flag = False
+
+    return fin_spec
+
+
+def saveData(data_obj, f_name):
+    """
+    Saves Lightcurve/EventList or any such data in chunks to disk.
+
+    Parameters
+    ----------
+    data_obj : :class:`stingray.Lightcurve` or :class:`stingray.events.EventList` object
+        Data to be stored on the disk.
+    f_name : string
+        Name of high level directory where data is to be stored
+
+    Raises
+    ------
+    ValueError
+        If data is not a Lightcurve or EventList
+    """
+    if isinstance(data_obj, stingray.Lightcurve):
+        _saveChunkLC(data_obj, f_name)
+
+    elif isinstance(data_obj, EventList):
+        _saveChunkEV(data_obj, f_name)
+
+    else:
+        raise ValueError(f"Cannot save data of type {type(data_obj).__name__}")
+
+
+def createChunkedSpectra(data_type, spec_type, segment_size, norm, gti, power_type, silent, dt):
+    """
+    Create a chunked spectra from zarr files stored on disk.
+
+    Parameters
+    ----------
+    data_type : string
+        Data in Lightcurve or EventList
+    spec_type : string
+        Type of spectra to create AveragedCrossspectrum or AveragedPowerspectrum.
+    segment_size: float
+        The size of each segment to average.
+    norm : {``frac``, ``abs``, ``leahy``, ``none``}
+        The normalization of the (real part of the) cross spectrum.
+    gti : 2-d float array
+        `[[gti0_0, gti0_1], [gti1_0, gti1_1], ...]`` -- Good Time intervals.
+        This choice overrides the GTIs in the single light curves. Use with
+        care!
+    power_type : string
+        Parameter to choose among complete, real part and magnitude of
+         the cross spectrum. None for AveragedPowerspectrum
+    silent : bool
+        Do not show a progress bar when generating an averaged cross spectrum.
+        Useful for the batch execution of many spectra
+    dt : float
+        The time resolution of the light curve. Only needed when constructing
+        light curves in the case where data1 or data2 are of :class:EventList
+
+    Returns
+    -------
+    object
+        Final computed spectra.
+    """
+    # Finds path to all stored zarr groups
+    data_path = sorted({
```

Comment:
Note that this is highly prone to race conditions. You are trusting the user to have one single zarr file during the processing, but this is clearly unwarranted. Please give a proper "file name" to the top of the zarr data directory tree

Reply:
Will do!

Reply:
In the saveDataEV and saveDataLC functions, I have set overwrite to True and hence even if there are existing zarr files saved it will overwrite them

Reply:
But I might want to analyze multiple datasets at once, or want to save the zarr datasets for further investigation without overwriting them :)

Reply:
Ohh okay I will give them generic names and check them later while reading them from the path